src/kOS.Safe/Encapsulation/Suffixes/NoArgsSuffix.cs

```diff
@@ -1,19 +1,42 @@
+using System;
+
 namespace kOS.Safe.Encapsulation.Suffixes
 {
-    public class NoArgsSuffix<TReturn> : SuffixBase
+    public class NoArgsSuffix<TReturn> : SuffixBase where TReturn : Structure
     {
         private readonly Del<TReturn> del;
 
-        public delegate TInnerReturn Del<out TInnerReturn>();
+        public delegate TInnerReturn Del<out TInnerReturn>() where TInnerReturn : Structure;
 
         public NoArgsSuffix(Del<TReturn> del, string description = ""):base(description)
         {
             this.del = del;
         }
 
-        public override object Get()
+        public override ISuffixResult Get()
+        {
+            return new DeletageSuffixResult<TReturn>(del);
+        }
+    }
+
+    public class DeletageSuffixResult<TReturn> : ISuffixResult where TReturn : Structure
+    {
+        private Delegate del;
+
+        public DeletageSuffixResult(Delegate del)
+        {
+            this.del = del;
+        }
+
+        public Structure Value()
+        {
+            //TODO:ERENDRAKE Make custom error
+            throw new System.NotImplementedException();
```

Comment:
My TODO on this one is incorrect. I believe that the delegate code currently living in OpcodeCall should be moved into here. Maybe.